portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -1462,7 +1463,8 @@ public String getCanonicalURL(
 
 			if ((pos <= 0) || (pos >= groupFriendlyURL.length())) {
 				sb.append(groupFriendlyURL);
-				sb.append(_buildI18NPath(siteDefaultLocale, layout.getGroup()));
```

Comment:
I think this will break canonical URLs. @austinchiang Do we have any specific ci:test:* command to check SEO/canonical URLs?